### PR TITLE
Improved adding relations in outlook

### PR DESCRIPTION
**Pull request by poradz**
image

### CODE REVIEW

File: public_html/layouts/basic/modules/MailIntegration/resources/Detail.js [hunk 1]
```diff
@@ -49,8 +49,12 @@ const MailIntegration_Detail = {
 			record: recordData.id,
 			recordModule: recordData.module
 		})
-			.done(function(responseData) {
-				console.info(responseData);
+			.done(response => {
+				Vtiger_Helper_Js.showPnotify({
+					text: app.vtranslate('JS_REMOVED_RELATIONS_SUCCESSFULLY'),
+					type: response['success'] ? 'success' : 'error'
```
Comment (rskrzypczak) [inline public_html/layouts/basic/modules/MailIntegration/resources/Detail.js]: JS_REMOVED_RELATIONS_SUCCESSFULLY and 'success' or'error'  ?

File: public_html/layouts/basic/modules/MailIntegration/resources/Detail.js [hunk 2]
```diff
@@ -61,18 +65,32 @@ const MailIntegration_Detail = {
...
+			module: 'MailIntegration',
+			action: 'Mail',
+			mode: 'addRelation',
+			mailId: this.mailId,
+			record: recordId,
+			recordModule: moduleName
+		}).done(response => {
+			Vtiger_Helper_Js.showPnotify({
+				text: app.vtranslate('JS_ADDED_RELATION_SUCCESSFULLY'),
+				type: response['success'] ? 'success' : 'error'
```
Comment (rskrzypczak) [inline public_html/layouts/basic/modules/MailIntegration/resources/Detail.js]: 'success' or'error' ?

File: public_html/layouts/basic/modules/MailIntegration/resources/Detail.js [hunk 3]
```diff
@@ -11,6 +11,21 @@ const MailIntegration_Detail = {
 		blockInfo: { enabled: true },
 		message: false
 	},
+	showResponseMessage(success, successLabel) {
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/MailIntegration/resources/Detail.js]:
```suggestion
	showResponseMessage(success, text) {
```

File: languages/en-US/_Base.json [hunk 4]
```diff
@@ -1825,6 +1825,10 @@
     "JS_EMOJI_FLAGS": "flags",
     "JS_EMOJI_CUSTOM": "custom",
     "JS_EMOJI_SEARCH": "Search…",
-    "JS_CANT_CLOSE_NEW_RECROD": "You cannot close a new ticket"
+    "JS_CANT_CLOSE_NEW_RECROD": "You cannot close a new ticket",
+    "JS_MAILINTEGRATION_ERROR": "YetiForce message: add-in error, please contact system administrator.",
```
Comment (mariuszkrzaczkowski) [inline languages/en-US/_Base.json]: languages/en-US/MailIntegration.json

File: public_html/layouts/basic/modules/MailIntegration/resources/Detail.js [hunk 5]
```diff
@@ -61,18 +77,29 @@ const MailIntegration_Detail = {
...
 		return false;
 	},
-	showQuickCreateForm(moduleName, relatedParams = {}) {
-		relatedParams['relationOperation'] = true;
-		const quickCreateParams = {
-			data: relatedParams,
-			noCache: true,
-			showInIframe: true
-		};
+	addRelation({ recordId, moduleName }) {
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/MailIntegration/resources/Detail.js]:
```suggestion
	addRelation(recordId, moduleName) {
```

File: public_html/layouts/basic/modules/MailIntegration/resources/Detail.js [hunk 6]
```diff
@@ -61,18 +77,29 @@ const MailIntegration_Detail = {
...
+			mailId: this.mailId,
+			record: recordId,
+			recordModule: moduleName
+		}).done(response => {
+			this.showResponseMessage(response['success'], 'JS_MAILINTEGRATION_ADDED_RELATION_SUCCESSFULLY');
+			this.reloadView(response['success']);
+		});
+	},
+	showQuickCreateForm(moduleName, quickCreateParams = {}) {
+		quickCreateParams = Object.assign({ noCache: true, showInIframe: true, data: {} }, quickCreateParams);
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/MailIntegration/resources/Detail.js]: ?

File: public_html/layouts/basic/modules/MailIntegration/resources/Detail.js [hunk 7]
```diff
@@ -194,27 +211,9 @@ const MailIntegration_Detail = {
...
-							text: response['data'],
-							animation: 'show'
-						};
-						if (response['success']) {
-							notifyParams.type = 'info';
-						}
-						this.iframeWindow.Vtiger_Helper_Js.showPnotify(notifyParams);
-						this.loadRelationsList();
-					});
+					this.addRelation({ recordId: responseData.id, moduleName: params.module });
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/MailIntegration/resources/Detail.js]:
```suggestion
					this.addRelation(responseData.id, params.module);
```

File: public_html/layouts/basic/modules/MailIntegration/resources/Detail.js [hunk 8]
```diff
@@ -228,18 +227,17 @@ const MailIntegration_Detail = {
 	},
 	registerAddRecord() {
 		this.addRecordBtn.on('click', e => {
-			this.showQuickCreateForm(this.moduleSelect[0].value);
+			const moduleName = this.moduleSelect[0].value;
+			const callbackFunction = ({ result }) => {
+				this.addRelation({ moduleName, recordId: result._recordId });
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/MailIntegration/resources/Detail.js]:
```suggestion
				this.addRelation(moduleName, result._recordId);
```

**Pull request merged**